core/src/main/kotlin/io/rover/campaigns/core/tracking/AutoScreenTracker.kt

```diff
@@ -0,0 +1,50 @@
+package io.rover.campaigns.core.tracking
+
+import android.app.Activity
+import android.app.Application
+import android.content.pm.PackageManager
+import android.os.Bundle
+import io.rover.campaigns.core.RoverCampaigns
+import io.rover.campaigns.core.events.EventQueueServiceInterface
+import io.rover.campaigns.core.logging.log
+import java.lang.Exception
+
+private const val TM_PACKAGE_PREFIX = "com.ticketmaster"
+private const val ROVER_PACKAGE_PREFIX = "io.rover"
+
+const val TRACKING_LABEL_KEY = "rvAutoTrackingLabelKey"
+const val AUTO_TRACKING_ENABLED_KEY = "rvAutoTrackingEnabled"
+const val ACTIVITY_EXCLUDE_FROM_TRACKING = "rvAutoTrackingExcludeActivity"
+
+internal class AutoScreenTracker : Application.ActivityLifecycleCallbacks {
+    override fun onActivityPaused(activity: Activity?) {}
+
+    override fun onActivityResumed(activity: Activity?) {
+        activity?.let {
+            try {
+                val activityInfo = it.packageManager?.getActivityInfo(it.componentName, PackageManager.GET_META_DATA)
```

Comment:
Can you get a reference to the actual Class of the dev's activity here?

On iOS we're gonna do a Protocol (akin to Interface) that will allow the dev to set contentID and contentName dynamically (at least those, haven't solidified it yet), so maybe for completeness we should offer the same feature here.